tests/bsim/bluetooth/audio/src/pacs_notify_client_test.c

```diff
@@ -0,0 +1,580 @@
+/*
+ * Copyright (c) 2023 Demant A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/gatt.h>
+
+#include "common.h"
+#include "common/bt_str.h"
+
+struct pacs_instance_t {
+	uint16_t start_handle;
+	uint16_t end_handle;
+
+	struct bt_gatt_subscribe_params sink_pacs_sub;
+	struct bt_gatt_subscribe_params source_pacs_sub;
+	struct bt_gatt_subscribe_params sink_loc_sub;
+	struct bt_gatt_subscribe_params source_loc_sub;
+	struct bt_gatt_subscribe_params available_contexts_sub;
+	struct bt_gatt_subscribe_params supported_contexts_sub;
+
+	struct bt_gatt_discover_params discover_params;
+};
+
+extern enum bst_result_t bst_result;
+
+CREATE_FLAG(flag_is_encrypted);
+CREATE_FLAG(flag_pacs_snk_discovered);
+CREATE_FLAG(flag_pacs_src_discovered);
+CREATE_FLAG(flag_snk_loc_discovered);
+CREATE_FLAG(flag_src_loc_discovered);
+CREATE_FLAG(flag_available_contexts_discovered);
+CREATE_FLAG(flag_supported_contexts_discovered);
+CREATE_FLAG(flag_all_notifications_received);
+
+static struct bt_uuid_16 uuid = BT_UUID_INIT_16(0);
+
+static volatile size_t num_notifications;
+
+static struct pacs_instance_t pacs_instance;
+
+static void security_changed(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	if (err) {
+		FAIL("Failed to change security (err %d)\n", err);
+		return;
+	}
+
+	if (level != BT_SECURITY_L2) {
+		FAIL("Wrong security level set (level %d)", level);
+		return;
+	}
+
+	SET_FLAG(flag_is_encrypted);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
```

Comment:
Please don't use `BT_CONN_CB_DEFINE` for the audio Babblesim tests, as this will always be registered regardless of whether we are running the PACS tests. Consider registering dynamically if you cannot use the common `connected` callback